k8s-operator/apis/v1alpha1/types_connector.go

```diff
@@ -41,16 +44,28 @@ type ConnectorList struct {
 	Items []Connector `json:"items"`
 }
 
-// ConnectorSpec defines the desired state of a ConnectorSpec.
+// ConnectorSpec describes the desired Tailscale component (subnet router or
+// exit node). A Connector instance must define exactly one  component.
+// +kubebuilder:validation:XValidation:rule="[has(self.subnetRouter), has(self.exitNode)].exists_one(x,x)",message="Exactly one of {subnetRouter, exitNode} must be defined."
+// +kubebuilder:validation:XValidation:rule="has(self.subnetRouter) == has(oldSelf.subnetRouter)",message="Subnetrouter cannot be set or unset after creating Connector. Define another Connector instance with the desired configuration."
+// +kubebuilder:validation:XValidation:rule="has(self.exitNode) == has(oldSelf.exitNode)",message="Exit node cannot be set or unset after creating Connector. Define another Connector instance with the desired configuration."
```

Comment:
These validations run at apply time and ensure that:
- exactly one of exit node/subnet router must be defined
- a Connector that previously defined exit node cannot be changed to subnet router and vice versa

(Presumably changing from one to the other is not a required flow and saves us from complicated cleanup attempts)